source/_posts/2017-03-01-Serpent-more-than-just-another-JSON-mapping-framework.md

```diff
@@ -0,0 +1,74 @@
+---
+title:  Serpent - more than just another JSON mapping framework
+date: 2017-03-01 10:10:57
+tags: ios,JSON,mapping,parsing,framework
+authorIds: 
+- maco
+categories:
+- iOS
+---
+
+<p align=center><img src="https://d1gwekl0pol55k.cloudfront.net/image/nstack/translate_values/Serpent_icon_MKPKLwKrTM.png" width=30%/></p>
+
+At [Nodes](https://nodesagency.com), we develop apps that rely a lot on complex REST APIs. The iOS team needed a fast and fully featured JSON mapping framework.
+We had been using our own internal parsing framework for Objective-C for a long time, but when Swift came out we decided to create a new one, in Swift, taking advantage of protocol-oriented programming. 
+
+After many iterations and a renaming (our JSON mapping framework was previously known as Serializable), we are proud to announce that [Serpent](https://github.com/nodes-ios/Serpent) has reached version 1.0.
+
+
+
+### Another JSON mapping framework?
+
+More than that. Serpent is [one of the fastest JSON mappers](https://github.com/nodes-ios/SerpentPerformanceComparison#-the-results)  out there and has the most features, according to [our analysis](https://github.com/nodes-ios/SerpentPerformanceComparison#-feature-comparison).
+
+<p align=center><img src = "https://raw.githubusercontent.com/nodes-ios/SerpentPerformanceComparison/master/chart.png"/></p>
+
+And maybe you only have to parse and map small JSONs, and you don't care that much about the time it takes. But what about the time it takes to write the mapping code? 
+
+We made a tool that goes hand in hand with Serpent: with <img src="https://d1gwekl0pol55k.cloudfront.net/image/nstack/translate_values/ModelBoiler_1281_Hkq7ey3ZGo.png" width=4%/> **Model Boiler** you only have to declare the properties on your model, and the parsing code will be generated for you, potentially saving you hours of work. 
+
+<p align=center><img src = "https://d1gwekl0pol55k.cloudfront.net/image/nstack/translate_values/ModelBoiler_5S0fTmSARI.gif"/></p>
+
+
+
+
+### But why build a new one?
+Yes, there are lots of other JSON mapping frameworks out there. But when we started developing Serpent, there was only SwiftyJSON, which wasn't exactly what we wanted. So we needed to build our own. 
+
+Since February 2016 we moved the project to a [public GitHub repo](https://github.com/nodes-ios/Serpent) and kept developing in the open. In the meantime, we were already using this framework in all our apps. At the moment, we can say that ***various versions of Serpent are part of apps with more than 1.3 million monthly users***.
```

Comment:
In February 2016, we moved the project to a [public GitHub repo](https://github.com/nodes-ios/Serpent) and began developing in the open. At the same time, we started using this framework in all our apps, and at the moment, we can say that ***Serpent is used in apps with more than 1.3 million monthly users***.